components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/OauthTokenIssuer.java

```diff
@@ -20,17 +20,18 @@
 package org.wso2.carbon.identity.oauth2.token;
 
 import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.wso2.carbon.identity.oauth.common.exception.InvalidOAuthClientException;
 import org.wso2.carbon.identity.oauth2.authz.OAuthAuthzReqMessageContext;
 
 public interface OauthTokenIssuer {
 
-    String accessToken(OAuthTokenReqMessageContext tokReqMsgCtx) throws OAuthSystemException;
+    String accessToken(OAuthTokenReqMessageContext tokReqMsgCtx) throws OAuthSystemException, InvalidOAuthClientException;
```

Comment:
Why do we need to change the interface to accommodate InvalidOAuthClientException?